erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -291,6 +291,7 @@ def update_item(source, target, source_parent):
 		target.base_amount = (flt(source.qty) - flt(source.delivered_qty)) * flt(source.base_rate)
 		target.amount = (flt(source.qty) - flt(source.delivered_qty)) * flt(source.rate)
 		target.qty = flt(source.qty) - flt(source.delivered_qty)
```

Comment:
Delivered Qty should be updated based on "Stock Qty", you have to change `self.status_updater` dict in delivery_note.py

So, this line should be 
`target.qty = flt(source.qty) - (flt(source.delivered_qty)/flt(source.conversion_factor))`